runtime/etc/Makefile.mli-common

```diff
@@ -0,0 +1,96 @@
+# Copyright 2004-2019 Cray Inc.
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Makefile.mli-common
+#
+
+include $(CHPL_MAKE_HOME)/runtime/etc/Makefile.include
+
+# Set the CHPL_ZMQ_HOME variable.
+ifdef CHPL_ZMQ_HOME
+CHPL_MLI_INCS = -I$(CHPL_ZMQ_HOME)/include
+CHPL_ZMQ_LIB_HOME = $(CHPL_ZMQ_HOME)/lib
+CHPL_ZMQ_LIB_SEARCH_PATH = -L$(CHPL_ZMQ_LIB_HOME)
+endif
+
+# Get the Chapel launcher library directory.
+CHPL_LN_LIB_DIR = $(CHPL_MAKE_RUNTIME_LIB)/$(CHPL_MAKE_LAUNCHER_SUBDIR)
+
+# Variables to propagate to the sub-make session.
+TO_PROPAGATE = \
+	CHPL_MAKE_HOST_TARGET="--host" \
+  CHPL_MAKE_HOME="$(CHPL_MAKE_HOME)" \
+  COMP_GEN_MLI_EXTRA_INCLUDES="$(COMP_GEN_MLI_EXTRA_INCLUDES)" \
+  TMPBINNAME="$(TMPBINNAME)" \
+  BINNAME="$(BINNAME)" \
+	TMPSERVERNAME="$(TMPSERVERNAME)" \
+	SERVERNAME="$(SERVERNAME)" \
+  CHPLSRC="$(CHPLSRC)" \
+  LIBS="$(LIBS)" \
+  CHPL_CL_OBJS="$(CHPL_CL_OBJS)" \
+  CHPL_MAKE_IS_MLI="$(CHPL_MAKE_IS_MLI)" \
+  TMPDIRNAME="$(TMPDIRNAME)"
+
+.PHONY: buildServer
+buildServer: $(TMPSERVERNAME)
+
+.PHONY: runTagsCommand
+runTagsCommand:
+	$(TAGS_COMMAND)
+
+# Compile loose source files in CHPLUSEROBJ.
+.PHONY: buildUserObj
+buildUserObj:
+	$(foreach srcFile, $(CHPLUSEROBJ), \
+		$(CC) $(CHPL_MAKE_BASE_CFLAGS) $(GEN_CFLAGS) $(COMP_GEN_CFLAGS) \
+			-c -o $(srcFile) $(CHPL_RT_INC_DIR) $(srcFile).c \
```

Comment:
Something seems awry with this.  Has it been tested?  The comment says we're compiling "loose source files in CHPLUSEROBJ".  But the compilation command here implies that the components of `CHPLUSER_OBJ` are object files, based on the `-o $(srcfile)` option.  If that's so, then the use of the identifier `srcFile` for the components of `CHPLUSEROBJ` is misleading.  And whether the files in `CHPLUSEROBJ` really are object files or source files, simply appending a '.c' as `$(srcFile).c` won't work anyway.

Reply:
I will have to look at the other Makefiles that use this variable to figure out what's going on here - this is just an artifact from a previous translation.

Reply:
Ah, I took that line wholesale (sans formatting) from `runtime/etc/Makefile.exe`. Are we sure if the rules in `Makefile.exe` are correct?

Reply:
One would think so, but surprises are never-ending it seems.  :-)

Can you try exercising that logic, even just the Makefile.exe version of it?  I tried a simplistic test but that ended up with `CHPLUSEROBJ` empty, so this loop didn't do anything.

Reply:
@gbtitus

This line was introduced in `Makefile.exe` as of `909897ea2c30e2f8952504cf3c4bdec7604a07ae`. The log for that commit says: 

```
Note that this feature is only enabled when incremental compilation is enabled via the command line. Otherwise the entire thing works as it previously used to work.
```

So it is effectively a NOP in most cases.

Reply:
That's reassuring in an odd way, because it sure looks broken to me.  What might be best would be to check `CHPLUSEROBJ` and if it's not empty, issue an error saying that incremental compilation is not yet supported with multilocale libraries.  Then hope that people get incremental compilation working first along the regular path, at which point when they want it for multilocale libraries you can steal their code!

The alternative is simply to leave it as is and hope that when someone tries it, the failure mode is obvious.

Reply:
Thanks, I'll do that.